### ISSUE TITLE
Implement Gendox API integration and  Document Splitter job from external source

### ISSUE THREAD

**Issue by sekasx** [ctrl-space-labs/gendox-core#354]
## Description

Currently, Gendox has integrations where the users upload their files in to the Gendox platform, and the the files are processed. It is not always possible to upload the original sources. For example a Web Page can be dynamically generated, or a user might have TBs of documents.

So, we need to read files from external sources, like HTML files or files from custom APIs. This way we can set integrations with external systems and just index the document contents upon split, without the need to copy the original document in Gendox.

Under this ticker 2 new Document Content Readers will be added:
- HTML Reader
- Custom WP Reader

**In the Splitter Job:**
In the DocumentInstance entity a new column will be added, named `document_reader_type` to indicate what reader should be used. Also the `remote_url` will be actually used to indicate external files. Upon Split, this reader will be used on `public String readDocumentContent(String documentUrl)` to return the document text (the method shouldn't called `download` anymore) 

**In the Integration:**
A new integration type should be added -> Remote Content Interation 
This integration is an experimental type trying to generalize the integration of gendox with external documents.
This integration will read the contents of a remote "repository", like, and FTP Server, a Web Page, a local or remote file system. The new/update/deleted files will be added in a temp table and then the contents of the DocumentInstance will be update.

### TL;DR
1. The Integration:
- New integration type connected to a remote System
- Finds the new/update/deleted files in a temp table
- Createds/Updates the DocumentInstance by setting the Remote URL of the file, and the `update_at`
- Deletes the missing files

2. The Splitter Job:
- Reads the updated DocumentInstances
- Gets the contents of the remote file
- Splits it, as per normal flow!
